### DISCUSSION TITLE
tiff2pdf error - No support for tif file with sample format 2

### DISCUSSION THREAD

**Discussion by priyankatapar** [libvips/libvips#2014] [Q&A]
### Summary

While converting some tif files to pdf getting error - "No support for tif file  with sample format 2."
Attaching the sample file -[multi-channel-z-series.ome.tif](/uploads/04bd3a7d6bdde38d8f4d8067f7b315ad/multi-channel-z-series.ome.tif)

### Version

LIBTIFF, Version 4.0.10

### Steps to reproduce

` # tiff2pdf -o out.pdf ./multi-channel-z-series.ome.tif
tiff2pdf: No support for ./multi-channel-z-series.ome.tif with sample format 2.
tiff2pdf: An error occurred creating output PDF file.`

On doing tiffinfo on that file:

`/ # tiffinfo ./multi-channel-z-series.ome.tif
/ # tiffinfo ./multi-channel-z-series.ome.tif
TIFF Directory at offset 0x8 (8)
  Image Width: 439 Image Length: 167
  Resolution: 0, 0 pixels/cm
  Bits/Sample: 8
  Sample Format: signed integer
  Compression Scheme: None
  Photometric Interpretation: min-is-black
  Samples/Pixel: 1
  Rows/Strip: 1
  Planar Configuration: single image plane
  ImageDescription: <?xml version="1.0" encoding="UTF-8"?><OME xmlns="http://www.openmicroscopy.org/Schemas/OME/2016-06" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" Creator="OME Bio-Formats 5.2.2" UUID="urn:uuid:bc26e771-02ad-4473-9f75-ae454861b991" xsi:schemaLocation="http://www.openmicroscopy.org/Schemas/OME/2016-06 http://www.openmicroscopy.org/Schemas/OME/2016-06/ome.xsd"><Image ID="Image:0" Name="multi-channel-z-series.ome.tif"><Pixels BigEndian="true" DimensionOrder="XYZCT" ID="Pixels:0" SizeC="3" SizeT="1" SizeX="439" SizeY="167" SizeZ="5" Type="int8"><Channel ID="Channel:0:0" SamplesPerPixel="1"><LightPath/></Channel><Channel ID="Channel:0:1" SamplesPerPixel="1"><LightPath/></Channel><Channel ID="Channel:0:2" SamplesPerPixel="1"><LightPath/></Channel><TiffData FirstC="0" FirstT="0" FirstZ="0" IFD="0" PlaneCount="1"><UUID FileName="multi-channel-z-series.ome.tif">urn:uuid:bc26e771-02ad-4473-9f75-ae454861b991</UUID></TiffData><TiffData FirstC="0" FirstT="0" FirstZ="1" IFD="1" PlaneCount="1"><UUID FileName="multi-channel-z-series.ome.tif">urn:uuid:bc26e771-02ad-4473-9f75-ae454861b991</UUID></TiffData><TiffData FirstC="0" FirstT="0" FirstZ="2" IFD="2" PlaneCount="1"><UUID FileName="multi-channel-z-series.ome.tif">urn:uuid:bc26e771-02ad-4473-9f75-ae454861b991</UUID></TiffData><TiffData FirstC="0" FirstT="0" FirstZ="3" IFD="3" PlaneCount="1"><UUID FileName="multi-channel-z-series.ome.tif">urn:uuid:bc26e771-02ad-4473-9f75-ae454861b991</UUID></TiffData><TiffData FirstC="0" FirstT="0" FirstZ="4" IFD="4" PlaneCount="1"><UUID FileName="multi-channel-z-series.ome.tif">urn:uuid:bc26e771-02ad-4473-9f75-ae454861b991</UUID></TiffData><TiffData FirstC="1" FirstT="0" FirstZ="0" IFD="5" PlaneCount="1"><UUID FileName="multi-channel-z-series.ome.tif">urn:uuid:bc26e771-02ad-4473-9f75-ae454861b991</UUID></TiffData><TiffData FirstC="1" FirstT="0" FirstZ="1" IFD="6" PlaneCount="1"><UUID FileName="multi-channel-z-series.ome.tif">urn:uuid:bc26e771-02ad-4473-9f75-ae454861b991</UUID></TiffData><TiffData FirstC="1" FirstT="0" FirstZ="2" IFD="7" PlaneCount="1"><UUID FileName="multi-channel-z-series.ome.tif">urn:uuid:bc26e771-02ad-4473-9f75-ae454861b991</UUID></TiffData><TiffData FirstC="1" FirstT="0" FirstZ="3" IFD="8" PlaneCount="1"><UUID FileName="multi-channel-z-series.ome.tif">urn:uuid:bc26e771-02ad-4473-9f75-ae454861b991</UUID></TiffData><TiffData FirstC="1" FirstT="0" FirstZ="4" IFD="9" PlaneCount="1"><UUID FileName="multi-channel-z-series.ome.tif">urn:uuid:bc26e771-02ad-4473-9f75-ae454861b991</UUID></TiffData><TiffData FirstC="2" FirstT="0" FirstZ="0" IFD="10" PlaneCount="1"><UUID FileName="multi-channel-z-series.ome.tif">urn:uuid:bc26e771-02ad-4473-9f75-ae454861b991</UUID></TiffData><TiffData FirstC="2" FirstT="0" FirstZ="1" IFD="11" PlaneCount="1"><UUID FileName="multi-channel-z-series.ome.tif">urn:uuid:bc26e771-02ad-4473-9f75-ae454861b991</UUID></TiffData><TiffData FirstC="2" FirstT="0" FirstZ="2" IFD="12" PlaneCount="1"><UUID FileName="multi-channel-z-series.ome.tif">urn:uuid:bc26e771-02ad-4473-9f75-ae454861b991</UUID></TiffData><TiffData FirstC="2" FirstT="0" FirstZ="3" IFD="13" PlaneCount="1"><UUID FileName="multi-channel-z-series.ome.tif">urn:uuid:bc26e771-02ad-4473-9f75-ae454861b991</UUID></TiffData><TiffData FirstC="2" FirstT="0" FirstZ="4" IFD="14" PlaneCount="1"><UUID FileName="multi-channel-z-series.ome.tif">urn:uuid:bc26e771-02ad-4473-9f75-ae454861b991</UUID></TiffData></Pixels></Image></OME>
  Software: OME Bio-Formats 5.2.2
TIFF Directory at offset 0x124cc (74956)
  Image Width: 439 Image Length: 167
  Resolution: 0, 0 pixels/cm
  Bits/Sample: 8
  Sample Format: signed integer
  Compression Scheme: None
  Photometric Interpretation: min-is-black
  Samples/Pixel: 1
  Rows/Strip: 1
  Planar Configuration: single image plane
  ImageDescription: ImageJ=
hyperstack=true
images=15
channels=3
slices=5
frames=1
  Software: OME Bio-Formats 5.2.2
TIFF Directory at offset 0x24990 (149904)
  Image Width: 439 Image Length: 167
  Resolution: 0, 0 pixels/cm
  Bits/Sample: 8
  Sample Format: signed integer
  Compression Scheme: None
  Photometric Interpretation: min-is-black
  Samples/Pixel: 1
  Rows/Strip: 1
  Planar Configuration: single image plane
  ImageDescription: ImageJ=
hyperstack=true
images=15
channels=3
slices=5
frames=1
  Software: OME Bio-Formats 5.2.2
TIFF Directory at offset 0x36e54 (224852)
  Image Width: 439 Image Length: 167
  Resolution: 0, 0 pixels/cm
  Bits/Sample: 8
  Sample Format: signed integer
  Compression Scheme: None
  Photometric Interpretation: min-is-black
  Samples/Pixel: 1
  Rows/Strip: 1
  Planar Configuration: single image plane
  ImageDescription: ImageJ=
hyperstack=true
images=15
channels=3
slices=5
frames=1
  Software: OME Bio-Formats 5.2.2
TIFF Directory at offset 0x49318 (299800)
  Image Width: 439 Image Length: 167
  Resolution: 0, 0 pixels/cm
  Bits/Sample: 8
  Sample Format: signed integer
  Compression Scheme: None
  Photometric Interpretation: min-is-black
  Samples/Pixel: 1
  Rows/Strip: 1
  Planar Configuration: single image plane
  ImageDescription: ImageJ=
hyperstack=true
images=15
channels=3
slices=5
frames=1
  Software: OME Bio-Formats 5.2.2
TIFF Directory at offset 0x5b7dc (374748)
  Image Width: 439 Image Length: 167
  Resolution: 0, 0 pixels/cm
  Bits/Sample: 8
  Sample Format: signed integer
  Compression Scheme: None
  Photometric Interpretation: min-is-black
  Samples/Pixel: 1
  Rows/Strip: 1
  Planar Configuration: single image plane
  ImageDescription: ImageJ=
hyperstack=true
images=15
channels=3
slices=5
frames=1
  Software: OME Bio-Formats 5.2.2
TIFF Directory at offset 0x6dca0 (449696)
  Image Width: 439 Image Length: 167
  Resolution: 0, 0 pixels/cm
  Bits/Sample: 8
  Sample Format: signed integer
  Compression Scheme: None
  Photometric Interpretation: min-is-black
  Samples/Pixel: 1
  Rows/Strip: 1
  Planar Configuration: single image plane
  ImageDescription: ImageJ=
hyperstack=true
images=15
channels=3
slices=5
frames=1
  Software: OME Bio-Formats 5.2.2
TIFF Directory at offset 0x80164 (524644)
  Image Width: 439 Image Length: 167
  Resolution: 0, 0 pixels/cm
  Bits/Sample: 8
  Sample Format: signed integer
  Compression Scheme: None
  Photometric Interpretation: min-is-black
  Samples/Pixel: 1
  Rows/Strip: 1
  Planar Configuration: single image plane
  ImageDescription: ImageJ=
hyperstack=true
images=15
channels=3
slices=5
frames=1
  Software: OME Bio-Formats 5.2.2
TIFF Directory at offset 0x92628 (599592)
  Image Width: 439 Image Length: 167
  Resolution: 0, 0 pixels/cm
  Bits/Sample: 8
  Sample Format: signed integer
  Compression Scheme: None
  Photometric Interpretation: min-is-black
  Samples/Pixel: 1
  Rows/Strip: 1
  Planar Configuration: single image plane
  ImageDescription: ImageJ=
hyperstack=true
images=15
channels=3
slices=5
frames=1
  Software: OME Bio-Formats 5.2.2
TIFF Directory at offset 0xa4aec (674540)
  Image Width: 439 Image Length: 167
  Resolution: 0, 0 pixels/cm
  Bits/Sample: 8
  Sample Format: signed integer
  Compression Scheme: None
  Photometric Interpretation: min-is-black
  Samples/Pixel: 1
  Rows/Strip: 1
  Planar Configuration: single image plane
  ImageDescription: ImageJ=
hyperstack=true
images=15
channels=3
slices=5
frames=1
  Software: OME Bio-Formats 5.2.2
TIFF Directory at offset 0xb6fb0 (749488)
  Image Width: 439 Image Length: 167
  Resolution: 0, 0 pixels/cm
  Bits/Sample: 8
  Sample Format: signed integer
  Compression Scheme: None
  Photometric Interpretation: min-is-black
  Samples/Pixel: 1
  Rows/Strip: 1
  Planar Configuration: single image plane
  ImageDescription: ImageJ=
hyperstack=true
images=15
channels=3
slices=5
frames=1
  Software: OME Bio-Formats 5.2.2
TIFF Directory at offset 0xc9474 (824436)
  Image Width: 439 Image Length: 167
  Resolution: 0, 0 pixels/cm
  Bits/Sample: 8
  Sample Format: signed integer
  Compression Scheme: None
  Photometric Interpretation: min-is-black
  Samples/Pixel: 1
  Rows/Strip: 1
  Planar Configuration: single image plane
  ImageDescription: ImageJ=
hyperstack=true
images=15
channels=3
slices=5
frames=1
  Software: OME Bio-Formats 5.2.2
TIFF Directory at offset 0xdb938 (899384)
  Image Width: 439 Image Length: 167
  Resolution: 0, 0 pixels/cm
  Bits/Sample: 8
  Sample Format: signed integer
  Compression Scheme: None
  Photometric Interpretation: min-is-black
  Samples/Pixel: 1
  Rows/Strip: 1
  Planar Configuration: single image plane
  ImageDescription: ImageJ=
hyperstack=true
images=15
channels=3
slices=5
frames=1
  Software: OME Bio-Formats 5.2.2
TIFF Directory at offset 0xeddfc (974332)
  Image Width: 439 Image Length: 167
  Resolution: 0, 0 pixels/cm
  Bits/Sample: 8
  Sample Format: signed integer
  Compression Scheme: None
  Photometric Interpretation: min-is-black
  Samples/Pixel: 1
  Rows/Strip: 1
  Planar Configuration: single image plane
  ImageDescription: ImageJ=
hyperstack=true
images=15
channels=3
slices=5
frames=1
  Software: OME Bio-Formats 5.2.2
TIFF Directory at offset 0x1002c0 (1049280)
  Image Width: 439 Image Length: 167
  Resolution: 0, 0 pixels/cm
  Bits/Sample: 8
  Sample Format: signed integer
  Compression Scheme: None
  Photometric Interpretation: min-is-black
  Samples/Pixel: 1
  Rows/Strip: 1
  Planar Configuration: single image plane
  ImageDescription: ImageJ=
hyperstack=true
images=15
channels=3
slices=5
frames=1
  Software: OME Bio-Formats 5.2.2
`

Platform

Docker Alpine OS

Any specific reason why `Signed Int Sample format` files are not supported while converting to pdf?
Also, are they standard files and still supported? Any pointers how to handle such file, will be very helpful.

Thanks in advance!

**Comment by jcupitt** [comment]
Hello, `tiff2pdf` is not part of libvips, you'd need to ask there.

The libvips equivalent would be:

```
vips magicksave multi-channel-z-series.ome.tif x.pdf
```

Though I would not write as PDF myself

**Comment by priyankatapar** [comment]
Thanks @jcupitt , I have posted in libtiff. I asked here, in case I could get some pointers. Thanks again!
